src/ccs/search_state.py

```diff
@@ -48,11 +58,18 @@ def __init__(
         self.tallies = tallies
         self.or_specificities = or_specificities
         self.prop_accumulator = prop_accumulator
-        self.props = props  # TODO add all root-level props from dag!
+        self.props = props
         self.poisoned = poisoned
 
+        if len(props) == 0:
+            for field, new_value in self._augment(deque(), activate_root=True).items():
+                setattr(self, field, new_value)
```

Comment:
This could certainly be made cleaner by using a factory function, but this odd use of `_augment` as both an initialization step and as a "cloning" function makes it messy. Worth fixing before merging, in my opinion.

Reply:
I agree this has gotten a little nasty. I'd rather merge it first, though.